lib/collections.c

```diff
@@ -87,3 +88,33 @@ void queue_free(Queue* queue)
     }
     free(queue);
 }
+
+// Conch ADT
+typedef struct Conch
+{
+    int64_t conch_value;
+    bool conch_available;
+} Conch;
+
+Conch* conch_init(int64_t initial_conch_value)
+{
+    Conch* conch = malloc(sizeof(Conch));
+    conch->conch_value = initial_conch_value;
+    return conch;
+}
+
+void conch_checkin(Conch* conch, int64_t new_conch_value)
+{
+    conch->conch_value = new_conch_value;
+    conch->conch_available = true;
+}
+
+int64_t conch_checkout(Conch* conch)
+{
```

Comment:
re: API design, I like these methods -- they feel like the "C way" of doing things.

I also really like asserting preconditions for stuff like this; e.g.
```
assert(conch.conch_available);
```
(and the opposite assertion in `conch_checkin`)

This should make it easier to debug if you *do* end up mis-using the API somehow.